drivers/flash/flash_stm32_ospi.c

```diff
@@ -1189,6 +1199,7 @@ static int flash_stm32_ospi_write(const struct device *dev, off_t addr,
 		ret = stm32_ospi_write_enable(&dev_data->hospi,
 						    dev_cfg->data_mode, dev_cfg->data_rate);
 		if (ret != 0) {
+			ospi_unlock_thread(dev);
```

Comment:
Is it really needed ?
 when breaking the  
`ospi_unlock_thread(dev);`  comes just before the 	`return ret;`   out of the while loop

Reply:
I will fix this